spec/ruby/language/predefined/toplevel_binding_spec.rb

```diff
@@ -0,0 +1,24 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+describe "The TOPLEVEL_BINDING constant" do
+  before :each do
+    @binding_toplevel_variables = eval(ruby_exe(fixture(__FILE__, "toplevel_binding_variables.rb")))
+    @binding_toplevel_id = eval(ruby_exe(fixture(__FILE__, "toplevel_binding_id.rb")))
+  end
+  
+  it "includes local variables defined in the same file" do
+    @binding_toplevel_variables.should include(:foo)
+  end
+
+  it "does not include local variables defined in a required file" do
+    @binding_toplevel_variables.should_not include(:bar)
+  end
+
+  it "does not include local variables defined in eval" do
+    @binding_toplevel_variables.should_not include(:baz)
+  end
```

Comment:
Isn't simply that the TOPLEVEL_BINDING is built as if there was `TOPLEVEL_BINDING = binding` in the main script file (the filename passed to `ruby`)?
We had a node in the past setting up TOPLEVEL_BINDING once.

Reply:
If it's set in the main script then what is it set to for files loaded via `-rfoo`?

Reply:
It seems the Binding is already created at that point but with no local variables:
```
$ ruby -r./sub main.rb
# in sub.rb
47001020681960
[]
# in main.rb
47001020681960
[:a, :b]
```

If we use `require_relative 'sub'` in main.rb, then locals of the main script (main.rb) are visible to both.
```
# in sub.rb
47090538696420
[:a, :b]
# in main.rb
47090538696420
[:a, :b]
```